### ISSUE TITLE
onlyHasTabIndex elements are completely ignored

### ISSUE THREAD

**Issue by Rogach** [philc/vimium#3508]
**Describe the bug**
Before commit caf13c2d elements that only had `tabindex` attribute on them were included in link hints. After that commit they are always ignored due to [this if statement](https://github.com/philc/vimium/blob/master/content_scripts/link_hints.coffee#L845):

```coffeescript
if visibleElement.secondClassCitizen
  continue
```

The commit in question is from PR #2251, it was related to hiding links to elements that are obscured by other elements. It feels like the dropping of all `tabindex` elements was not intentional.

(quote from code before the change: "we don't want to stop being able to click some elements that we could click before" - I wholeheartedly agree)

**To Reproduce**
Here's [the codepen with a single div with tabindex](https://codepen.io/Rogach/pen/BaNWwLM) for convenient testing.

**Browser and Vimium version**
Browser: Chromium 79.0.3945.117 (Official Build) Arch Linux (64-bit)
OS: Linux
Vimium: 1.65.2

**Comment by everyonesdesign** [comment]
I see that there is intention of treating tabindex-only element in a different way ([source](https://github.com/philc/vimium/blob/master/content_scripts/link_hints.coffee#L754)). 

But I wonder, are there many cases when making tabindex element clickable leads to something undesirable. It would be interesting to listen to input from the devs on the rationales behind this decision.

**Comment by lydell** [comment]
> are there many cases when making tabindex element clickable leads to something undesirable

Good question! It’s also valid to ask the opposite question: Do you have examples of sites where looking at tabindex helps?

**Comment by Rogach** [comment]
Well, I actually stumbled upon this because vimium navigation became broken in one of my personal apps after tabindex became unused. But unfortunately it's not available on public network.

**Comment by lydell** [comment]
If it’s your own app you have the possibility of improving the accessibility of it, I guess :)

**Comment by Rogach** [comment]
It's not exactly my own, but maybe I would be able to find some way to fix it. However I'm not sure exactly how - I was not able to find a full description of what vimium considers a "eligible" element anywhere (README or the wiki), so I need to rely on sources. Maybe there is such description somewhere?

**Comment by lydell** [comment]
Well, if you make the button/link discoverable as clickable for assistive technology (such as screen readers) there’s a good chance Vimium will find it. If it changes the URL, use `<a>`, otherwise a `<button>`. If that’s not possible, maybe add `role="button"`.

**Comment by Rogach** [comment]
Yes, it seems `role="button"` should work: https://github.com/philc/vimium/blob/master/content_scripts/link_hints.coffee#L689

**Comment by everyonesdesign** [comment]
For me it was also only a custom page that was broken. But today I've noticed that courses on Duolingo are not accessible for clicking via Vimium: 

![image](https://user-images.githubusercontent.com/3520704/75623901-5879c080-5baf-11ea-9dba-646c4f8b2f72.png)

**Comment by Rogach** [comment]
Can some of the developers comment on the desirability of `tabindex` elements? If they are not desirable I can provide a PR that will remove unnecessary code, otherwise I can provide a PR that will fix selection of such elements. Don't want to leave these issue hanging.

**Comment by andryandrew** [comment]
For what's worth, Duolingo's lessons not being clickable affects me too and I guess I would want to have the possibility back. But I would understand if this goes against future development directions...

**Comment by sersorrel** [comment]
This breaks the ability to click on the "next video" button on [panopto](https://www.panopto.com/), which has a `tabindex=0` and a `click` event handler. Replacing `onlyHasTabIndex` with `possibleFalsePositive` on [line 999 of link_hints.js](https://github.com/philc/vimium/blob/bdf654aebe6f570f427c5f7bc9592cad86e642b5/content_scripts/link_hints.js#L999) appears to be an ok interim fix.

**Comment by GreenBlast** [comment]
Is there a workaround possible for duolingo?

**Comment by andryandrew** [comment]
I switched to [vimium-c](https://github.com/gdh1995/vimium-c), which works correctly on Duolingo
